### ISSUE TITLE
Please help: Error: Disagreeing option values for '--link': 'logistic' vs 'identity'

### ISSUE THREAD

**Issue by Alluxia-F** [VowpalWabbit/vowpal_wabbit#1824]
#### 
vw --version: 8.6.1, installed with: brew install vowpal-wabbit
Computer: mac OS Mojave Version 10.14.4
#### 
Model training successfully. But when I run the following code for prediction, it shows the error. Detailed code as below:

$ vw -d test.vw -t -i neural_model --link=logistic -p probabilities.txt
creating quadratic features for pairs: sd ad do fd 
only testing
predictions = probabilities.txt
Error: Disagreeing option values for '--link': 'logistic' vs 'identity'
finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
Disagreeing option values for '--link': 'logistic' vs 'identity'
vw (parser_helper.cc:172): Disagreeing option values for '--link': 'logistic' vs 'identity'

#### Additional context
When I run the model training, it finished successfully. code as below:

$ vw -d train.vw --cache_file  neural --inpass --passes 5 -q sd  -q ad  -q do -q fd --binary -f neural_model  --loss_function=logistic  --nn 3

finished run
number of examples per pass = 36386071
passes used = 5
weighted example sum = 212824820.000000
weighted label sum = -89246960.000000
average loss = 0.269664 h
best constant = -0.893793
best constant's loss = 0.602445
total feature number = 21081101440

**Comment by jackgerrits** [comment]
Hi @Alluxia-F, can you please try adding `--link logistic` to your training command and then trying this scenario again to see if it fixes it?

So your training command should become:
```
$ vw -d train.vw --cache_file neural --inpass --passes 5 -q sd -q ad -q do -q fd --binary -f neural_model --loss_function=logistic --nn 3 --link logistic
```

**Comment by Alluxia-F** [comment]
@jackgerrits , thank you for the quick response. 

I tried to update the training command per suggestion above and then ran the following prediction command:
!vw -d test.vw -t -i neural_model -p nerural_model_prob.txt --link logistic

No probablitites generated in the  nerural_model_prob.txt file at all. All of the values are : 1.

creating quadratic features for pairs: sd ad do fd 
only testing
predictions = nerural_model_prob.txt
ignoring duplicate option: '--link logistic'
using input passthrough for neural network testing
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = test.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0   1.0000   1.0000      110
0.000000 0.000000            2            2.0   1.0000   1.0000      110
0.000000 0.000000            4            4.0   1.0000   1.0000      110
0.000000 0.000000            8            8.0   1.0000   1.0000      110
0.000000 0.000000           16           16.0   1.0000   1.0000      110
0.000000 0.000000           32           32.0   1.0000   1.0000      110
0.000000 0.000000           64           64.0   1.0000   1.0000      110
0.000000 0.000000          128          128.0   1.0000   1.0000      110
0.000000 0.000000          256          256.0   1.0000   1.0000      110
0.000000 0.000000          512          512.0   1.0000   1.0000      110
0.000000 0.000000         1024         1024.0   1.0000   1.0000      110
0.000000 0.000000         2048         2048.0   1.0000   1.0000      110
0.000000 0.000000         4096         4096.0   1.0000   1.0000      110
0.000000 0.000000         8192         8192.0   1.0000   1.0000      110
0.000000 0.000000        16384        16384.0   1.0000   1.0000      110
0.000000 0.000000        32768        32768.0   1.0000   1.0000      110
0.000000 0.000000        65536        65536.0   1.0000   1.0000      110
0.000000 0.000000       131072       131072.0   1.0000   1.0000      116
0.000000 0.000000       262144       262144.0   1.0000   1.0000      116
0.000000 0.000000       524288       524288.0   1.0000   1.0000      116
0.000000 0.000000      1048576      1048576.0   1.0000   1.0000      116
0.000000 0.000000      2097152      2097152.0   1.0000   1.0000      116
0.000000 0.000000      4194304      4194304.0   1.0000   1.0000      116

finished run
number of examples = 4577464
weighted example sum = 4577464.000000
weighted label sum = 4577464.000000
average loss = 0.000000
best constant = 1.000000
best constant's loss = 0.000000
total feature number = 530439884

**Comment by jackgerrits** [comment]
Are you able to provide the test and train files so we can look into this?

**Comment by arielf** [comment]
@Alluxia-F 

We need to see the data-set (or at least a sample of it) to fully understand what's going on, but based on the progress output, and in particular this part:
```
average loss = 0.000000
best constant = 1.000000
best constant's loss = 0.000000
```
It seems like the dominant label in the data-set is 1.0 so by predicting 1 always, the model gets perfect accuracy.

For more details and how to solve this problem see [this answer on Stackoverflow](https://stackoverflow.com/questions/24634602/how-to-perform-logistic-regression-using-vowpal-wabbit-on-very-imbalanced-datase/24641529#24641529)

**Comment by lokitoth** [comment]
Hi @Alluxia-F: Just wanted to loop back with you to see if your issue was resolved, or if you are still running into problem.

**Comment by jackgerrits** [comment]
Closing this as it is unfortunately unactionable without a repro.
